src/android/Capture.java

```diff
@@ -377,19 +381,60 @@ else if (resultCode == Activity.RESULT_CANCELED) {
         }
     }
 
-
     public void onAudioActivityResult(Request req, Intent intent) {
-        // Get the uri of the audio clip
-        Uri data = intent.getData();
-        if (data == null) {
+        Uri srcContentUri = intent.getData();
+
+        if (srcContentUri == null) {
             pendingRequests.resolveWithFailure(req, createErrorObject(CAPTURE_NO_MEDIA_FILES, "Error: data is null"));
             return;
         }
 
-        // Create a file object from the uri
-        JSONObject mediaFile = createMediaFile(data);
-        if (mediaFile == null) {
-            pendingRequests.resolveWithFailure(req, createErrorObject(CAPTURE_INTERNAL_ERR, "Error: no mediaFile created from " + data));
+        // Get file name
```

Comment:
What about not replacing but adding this new code inside the `if (mediaFile == null) {` as a "fallback"?
To not copy to and use the cacheDir if the captured audio file is already "correct".

My first tests with your POC went really well 👍!